.changeset/lucky-goats-kneel.md

```diff
@@ -0,0 +1,5 @@
+---
+'@itwin/itwinui-react': patch
+---
+
+Reduced layout thrashing on `Table` component by wrapping an expensive ref function in a callback.
```

Comment:
nit: "callback" is just a generic term; better to use the word "memoize". (`useCallback` is actually more like `useMemoizedCallback`)

```suggestion
Reduced layout thrashing on `Table` component by memoizing an expensive `ref` function.
```